HW14/part2/Box.java

```diff
@@ -0,0 +1,127 @@
+package part2;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class Box<T> {
+    private final List<T> fruitsBox;
+    private int numOfFruits;
+    boolean isEmpty;
+
+    public Box() {
+        fruitsBox = new ArrayList<>();
+        isEmpty = true;
+    }
+
+    @SafeVarargs
+    public Box(T... ts) {
+        fruitsBox = new ArrayList<>();
+        isEmpty = true;
+        addFew(ts);
+    }
+
+    public List<T> getFruitsBox() {
+        return fruitsBox;
+    }
+
+    // метод без аргументов просто удаляет
+    // метод, хотя бы с одним, аргументом, выводит сообщение
+    // когда коробка пуста
+    public void removeOne(String... strings) {
+        if (numOfFruits == 0) {
+            isEmpty = true;
+            if (strings.length >= 1) {
+                System.out.println("This box is empty.");
+            }
+        } else {
+            fruitsBox.remove(fruitsBox.size() - 1);
+            numOfFruits--;
+            if (numOfFruits == 0) {
+                isEmpty = true;
+            }
+        }
+    }
+
+    public void addOne(T t) {
+        if (numOfFruits == 0) {
+            fruitsBox.add(t);
+            numOfFruits++;
+            isEmpty = false;
+        } else {
+            if (fruitsBox.get(0).getClass().equals(t.getClass())) {
+                fruitsBox.add(t);
+                numOfFruits++;
+                isEmpty = false;
+            }
+        }
+    }
+
+    @SafeVarargs
+    public final void addFew(T... ts) {
+        for (T t : ts) {
+            if (numOfFruits == 0) {
+                fruitsBox.add(t);
+                numOfFruits++;
+                isEmpty = false;
+            } else {
+                if (fruitsBox.get(0).getClass().equals(t.getClass())) {
```

Comment:
I think this is the incorrect way to achieve the requirement of "the Box must be for apples to oranges only".
Generics functionality must resolve the "unique type issue" that you are solving via reflection.